### PR TITLE
Autogen rulename

### PR DESCRIPTION
**Pull request by GraysonWu**

### CODE REVIEW

File: pkg/controller/networkpolicy/mutate.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	switch op {
+	case admv1.Create, admv1.Update:
+		randStrLen := 8
+		ruleNames := sets.NewString()
+		listIntoSet := func(rules []secv1alpha1.Rule) {
+			for _, rule := range rules {
+				ruleNames.Insert(rule.Name)
+			}
+		}
+		listIntoSet(ingress)
```
Comment (Dyanngg) [inline pkg/controller/networkpolicy/mutate.go]: listIntoSet(append(ingress, egress...))

File: pkg/controller/networkpolicy/mutate.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+		listIntoSet(egress)
+
+		var paths []string
+		var values []string
+		prepareAutoNamePatch := func(prefix string, rules []secv1alpha1.Rule) {
```
Comment (Dyanngg) [inline pkg/controller/networkpolicy/mutate.go]: make this a standalone function that returns paths, values

File: pkg/controller/networkpolicy/mutate.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+					var genName string
+					for {
+						genName = fmt.Sprintf("%s-%s-%s", prefix, *rule.Action, randSeq(randStrLen))
+						if !ruleNames.Has(genName) {
+							ruleNames.Insert(genName)
+							break
+						}
+					}
+					paths = append(paths, fmt.Sprintf("/spec/%s/%d/name", prefix, idx))
+					values = append(values, genName)
```
Comment (Dyanngg) [inline pkg/controller/networkpolicy/mutate.go]:
```suggestion
					for {
						genName := fmt.Sprintf("%s-%s-%s", prefix, *rule.Action, randSeq(randStrLen))
						if !ruleNames.Has(genName) {
							ruleNames.Insert(genName)
							paths = append(paths, fmt.Sprintf("/spec/%s/%d/name", prefix, idx))
					                values = append(values, genName)
							break
						}
					}
```

File: pkg/controller/networkpolicy/mutate.go (hunk 1 above)
Comment (GraysonWu) [reply pkg/controller/networkpolicy/mutate.go]: As I discussed with quan here https://github.com/vmware-tanzu/antrea/pull/1330#discussion_r510973495 
I'm not sure if it's ok to lead one allocation?

Comment (Dyanngg) [reply pkg/controller/networkpolicy/mutate.go]: Sure, we can keep it this way

**Pull request closed without merging**